src/actor/Verifier.ts

```diff
@@ -0,0 +1,148 @@
+ */
+export class PresentationRequestBuilder {
+  private builder: gabi.CombinedRequestBuilder
+  private ctypes: Array<CType['hash']>
+  constructor() {
+    this.builder = new gabi.CombinedRequestBuilder()
+    this.ctypes = []
+  }
+
+  /**
+   * Initiates a verification by creating a presentation request for a specific [[CType]].
+   * Note that you are required to call [[finalize]] on the request to conclude it.
+   *
+   * @param p The parameter object.
+   * @param p.ctypehash The SHA-256 hash of the [[CType]].
+   * @param p.attributes A list of attributes of the [[Credential]]s the Verifier has to see in order to verify it.
+   * @param p.legitimations An optional boolean representing whether the Verifier requests to see the legitimations of the Attesters which signed the [[Credential]]s.
+   * @param p.delegationId An optional boolean representing whether the Verifier requests to see the Attesters' unique delegation identifiers.
+   * @param p.reqUpdatedAfter The optional minimum required timestamp on which the [[Credential]] needs to be updated.
+   * The default value for this is the current date.
+   * @returns A [[PresentationRequestBuilder]] on which you need to call [[finalize]] to complete the presentation request.
+   */
+  public requestPresentationForCtype({
+    ctypeHash,
+    attributes,
+    legitimations,
+    delegationId,
+    reqUpdatedAfter = new Date(),
+  }: {
+    attributes: string[]
+    ctypeHash: CType['hash']
+    legitimations?: boolean
+    delegationId?: boolean
+    reqUpdatedAfter?: Date
+  }): PresentationRequestBuilder {
+    const rawAttribute = attributes.map(attr => `claim.contents.${attr}`)
+    rawAttribute.push('claim.cTypeHash')
+    if (typeof legitimations !== 'undefined' && legitimations) {
+      rawAttribute.push('legitimation')
+    }
+    if (typeof delegationId !== 'undefined' && delegationId) {
+      rawAttribute.push('delegationId')
+    }
+    this.builder.requestPresentation({
+      requestedAttributes: rawAttribute,
+      reqUpdatedAfter,
+    })
+    this.ctypes.push(ctypeHash)
+    return this
+  }
+
+  /**
+   * [ASYNC] Concludes the presentation request.
+   *
+   * @param allowPE A boolean representing whether the verifier accepts a privacy enhanced presentation.
+   * @returns A session and a message object.
+   * The **session** object will be used in [[verifyPresentation]] and should be kept private by the verifier.
+   * The **message** object should be sent to the Claimer and used in [[createPresentation]].
+   */
+  public async finalize(
+    allowPE: boolean
+  ): Promise<[gabi.CombinedVerificationSession, IRequestClaimsForCTypes]> {
+    const { session, message } = await this.builder.finalise()
+    return [
+      session,
+      {
+        type: MessageBodyType.REQUEST_CLAIMS_FOR_CTYPES,
+        content: {
+          ctypes: this.ctypes,
+          peRequest: message,
+          allowPE,
```

Comment:
The claimer could possibly just return a pe version, while the verifier set false here.
He would have to save it seperately. Would be easier to just save it in the session

Reply:
fixed in b5823f0af7dfd1bbc76d9a66e73242a457b3550c